demisto_sdk/commands/common/hook_validations/pre_process_rule.py

```diff
@@ -0,0 +1,111 @@
+                error_message, error_code = Errors.invalid_from_server_version_in_pre_process_rules('fromServerVersion')
+                if self.handle_error(error_message, error_code, file_path=self.file_path):
+                    return False
+        return True
+
+    # # TODO Needed! Check that scripts and incident_fields exists
+    # # Needed incident_fields: description
+    # # If the key "scriptName" is not empty - Check that the script exists
+    # def is_incident_field_exist(self, id_set_file, is_ci) -> bool:
+    #     """Checks if incident field is valid - exist in the content.
+
+    #     Returns:
+    #         bool. True if incident field is valid, else False.
+    #     """
+    #     if not is_ci:
+    #         return True
+
+    #     if not id_set_file:
+    #         click.secho("Skipping mapper incident field validation. Could not read id_set.json.", fg="yellow")
+    #         return True
+
+    #     pre_process_rules_incident_fields = []
+
+    #     layout = self.current_file.get('layout', {})
+    #     pre_process_rules_sections = layout.get('sections', [])
+    #     for section in pre_process_rules_sections:
+    #         for field in section.get('fields', []):
+    #             inc_field = field.get('fieldId', '')
+    #             pre_process_rules_incident_fields.append(inc_field.replace('incident_', ''))
+
+    #     layout_tabs = layout.get('tabs', [])
+    #     for tab in layout_tabs:
+    #         pre_process_rules_sections = tab.get('sections', [])
+
+    #         for section in pre_process_rules_sections:
+    #             if section and section.get('items'):
+    #                 for item in section.get('items', []):
+    #                     inc_field = item.get('fieldId', '')
+    #                     pre_process_rules_incident_fields.append(inc_field.replace('incident_', '').replace('indicator_', ''))
+
+    #     content_incident_fields = get_all_incident_and_indicator_fields_from_id_set(id_set_file, 'layout')
+
+    #     built_in_fields = [field.lower() for field in BUILT_IN_FIELDS] + PRE_PROCESS_RULES_BUILT_IN_FIELDS
+
+    #     invalid_inc_fields_list = []
+    #     for inc_field in pre_process_rules_incident_fields:
+    #         if inc_field and inc_field.lower() not in built_in_fields and inc_field not in content_incident_fields:
+    #             invalid_inc_fields_list.append(inc_field) if inc_field not in invalid_inc_fields_list else None
+
+    #     if invalid_inc_fields_list:
+    #         error_message, error_code = Errors.invalid_incident_field_in_pre_process_rules(invalid_inc_fields_list)
+    #         if self.handle_error(error_message, error_code, file_path=self.file_path):
+    #             return False
+    #     return True
+
+    def is_script_exist(self) -> bool:
+        """Check if the file exists"""
+        if not os.path.isfile(self.file_path):
+            return False
+
+        return True
```

Comment:
this is not the right validation.

Reply:
https://github.com/demisto/demisto-sdk/commit/15720877ffa83827d9aa1a2060c9e5763e047df2